### DISCUSSION TITLE
cypher-gremlin-extensions?

### DISCUSSION THREAD

**Discussion by artturimatias** [ArcadeData/arcadedb#390] [Ideas]
Current OpenCypher implementation does not support string functions (toUpper, trim etc..)

`Custom functions and predicates are not supported on target implementation: cypherToUpper. Consider installing Extensions to Gremlin to enable full support for Cypher functionality: https://github.com/opencypher/cypher-for-gremlin/tree/master/tinkerpop/cypher-gremlin-extensions`

It seems that these functions require cypher-gremlin-extension to be present. Would it be possible to include these extensions to Docker image?

**Comment by lvca** [comment]
Does anyone know how to include those extensions?

**Reply by artturimatias** [reply]
Sorry, I really have no idea. Docs say: "The easiest way to use this module is by installing the [Gremlin Server Cypher plugin](https://github.com/opencypher/cypher-for-gremlin/blob/master/tinkerpop/cypher-gremlin-server-plugin) on the target Gremlin Server. The plugin includes all of the extensions and registers them on the Server."
https://github.com/opencypher/cypher-for-gremlin/tree/master/tinkerpop/cypher-gremlin-extensions
